cas-mfa-java/src/main/java/net/unicon/cas/mfa/authentication/DefaultRegisteredServiceMfaRoleProcessorImpl.java

```diff
@@ -170,7 +174,7 @@ private MultiFactorAuthenticationRequestContext getMfaRequestContext(final Servi
                     serviceMfaData.getAuthenticationMethod()).getRank();
             final MultiFactorAuthenticationSupportingWebApplicationService svc =
                     this.mfaServiceFactory.create(targetService.getId(), targetService.getId(),
-                            targetService.getArtifactId(), serviceMfaData.getAuthenticationMethod(),
+                            targetService.getArtifactId(), "POST".equals(method) ? ResponseType.POST : ResponseType.REDIRECT, serviceMfaData.getAuthenticationMethod(),
```

Comment:
And since it appears they really do need to create the service, how do we pass the response the service wants into the `mfaServiceFactory` from here?